### ISSUE TITLE
Problem to import Basemap (1.2.0) in Spyder - KeyError: 'PROJ_LIB' - Windows 10

### ISSUE THREAD

**Issue by rpedruzzi** [conda-forge/basemap-feedstock#37]
Issue:
Dear all,
I am using spyder 3.3.1 with python 3.6.5, Ipython 6.4.0 and I am having problem with Basemap. 
I tried to install the new anaconda with python 3.7 and I have the same problem.
I do not problems with other packages, like cartopy, which uses proj/proj.
Can you help me?
Below is the issue.

Traceback (most recent call last):

  File "<ipython-input-5-d9467465a3b6>", line 1, in <module>
    from mpl_toolkits.basemap import Basemap

  File "C:\Users\pedruzzi\anaconda3\lib\site-packages\mpl_toolkits\basemap\__init__.py", line 155, in <module>
    pyproj_datadir = os.environ['PROJ_LIB']

  File "C:\Users\pedruzzi\anaconda3\lib\os.py", line 669, in __getitem__
    raise KeyError(key) from None

KeyError: 'PROJ_LIB'

<br/>
Environment (<code>conda list</code>):
<details>
$ conda list
# packages in environment at C:\Users\pedruzzi\anaconda3:
#
# Name                       Version                   Build               Channel
basemap                        1.2.0            py36h5987640_0     conda-forge
basemap-data-hires      1.2.0                         0                  conda-forge
anaconda-project          0.8.2            py36hfad2e28_0
proj4                              4.9.3                hfa6e2cd_8            conda-forge
pyproj                             1.9.5.1                  py36_0             conda-forge

<br/>

Details about  <code>conda</code> and system ( <code>conda info</code> ):

$ conda info

     active environment : base
           conda version : 4.5.11
    conda-build version : 3.10.5
         python version : 3.6.5.final.0
                     platform : win-64
             user-agent : conda/4.5.11 requests/2.18.4 CPython/3.6.5 Windows/10 Windows/10.0.17134
          administrator : False
             netrc file : None
           offline mode : False

**Comment by ocefpaf** [comment]
Same as #30

**Comment by jamessong** [comment]
have you solved this?  @rpedruzzi

**Comment by rpedruzzi** [comment]
@jamessong,
Yeah, I changed the __init__ file of basemap.
this is what I did:
At the begging of the file, I added the os and conda libraries:
import os, conda
 
at the line 155, I commented the:
#pyproj_datadir = os.environ['PROJ_LIB']

and added the following lines:

if 'PROJ_DIR' in os.environ:
    pyproj_datadir = os.environ['PROJ_DIR']
else:
    conda_file_dir = conda.CONDA_PACKAGE_ROOT
    conda_dir = conda_file_dir.split('lib')[0]
    pyproj_datadir = os.path.join(os.path.join(conda_dir, 'Library'),'share')


The user just need to check the path to where the python system saved the epsg of pyproj/proj,
which is usually in Library/share or inside the the pyproj folder.

For the record, I'm using win10, anaconda 5.3.1, python 3.7.0 and I changed my first level channel to conda-forge.

regards

**Comment by jamessong** [comment]
thanks! Did you install anaconda with admin account or local user account? Somehow I can't modify that init file 

    On Monday, November 26, 2018 11:01 AM, Rizzieri Pedruzzi <notifications@github.com> wrote:
 

 @jamessong,
Yeah, I changed the init file of basemap.
this is what I did:
At the begging of the file, I added the os and conda libraries:
import os, condaat the line 155, I commented the:
#pyproj_datadir = os.environ['PROJ_LIB']and added the following lines:if 'PROJ_DIR' in os.environ:
pyproj_datadir = os.environ['PROJ_DIR']
else:
conda_file_dir = conda.CONDA_PACKAGE_ROOT
conda_dir = conda_file_dir.split('lib')[0]
pyproj_datadir = os.path.join(os.path.join(conda_dir, 'Library'),'share')The user just need to check the path to where the python system saved the epsg of pyproj/proj,
which is usually in Library/share or inside the the pyproj folder.For the record, I'm using win10, anaconda 5.3.1, python 3.7.0 and I changed my first level channel to conda-forge.regards—

**Comment by rpedruzzi** [comment]
@jamessong 

I'm in the admin group in my computer, so I have some permissions, even being a normal user. Additionally, I installed my python/anaconda inside my personal folder (e.g. C:\Users\your_user) instead in ~\AppData\, where usually anaconda install it. 
try to install your python/anaconda inside your user folder, like C:\Users\jamessong

**Comment by jamessong** [comment]
somehow I couldn't get spyder or jupyter notebook to work with it:

KeyError: 'PROJ_LIB'  

    On Monday, November 26, 2018 11:21 AM, Rizzieri Pedruzzi <notifications@github.com> wrote:
 

 @jamessongI'm in the admin group in my computer, so I have some permissions, even being a normal user. Additionally, I installed my python/anaconda inside my personal folder (e.g. C:\Users\your_user) instead in ~\AppData, where usually anaconda install it.
try to install your python/anaconda inside your user folder, like C:\Users\jamessong—

**Comment by Leasberthou** [comment]
Of all the solutions I found online the solution from @rpedruzzi was the only one that worked for me with the missing 'PROJ_LIB'. I had to reinstall Ananconda in a different path though.
For reference I use Anaconda 3 (version 2018.12) and Python 3.7.

**Comment by antonia-had** [comment]
> @jamessong,
> Yeah, I changed the **init** file of basemap.
> this is what I did:
> At the begging of the file, I added the os and conda libraries:
> import os, conda
> 
> at the line 155, I commented the:
> 
> ```python
> #pyproj_datadir = os.environ['PROJ_LIB']
> ```
> 
> and added the following lines:
> 
> ```python
> if 'PROJ_DIR' in os.environ:
>     pyproj_datadir = os.environ['PROJ_DIR']
> else:
>     conda_file_dir = conda.CONDA_PACKAGE_ROOT
>     conda_dir = conda_file_dir.split('lib')[0]
>     pyproj_datadir = os.path.join(os.path.join(conda_dir, 'Library'),'share')
> ```
> 
> The user just need to check the path to where the python system saved the epsg of pyproj/proj,
> which is usually in Library/share or inside the the pyproj folder.
> 
> For the record, I'm using win10, anaconda 5.3.1, python 3.7.0 and I changed my first level channel to conda-forge.
> 
> regards

Seconding that after having wasted a couple hours on this, the only solution that worked for me also, was @rpedruzzi's. I'm running on Windows 10, Python 3.6.

**Comment by DipankarMedhi** [comment]
> > @jamessong,
> > Yeah, I changed the **init** file of basemap.
> > this is what I did:
> > At the begging of the file, I added the os and conda libraries:
> > import os, conda
> > at the line 155, I commented the:
> > ```python
> > #pyproj_datadir = os.environ['PROJ_LIB']
> > ```
> > 
> > 
> > and added the following lines:
> > ```python
> > if 'PROJ_DIR' in os.environ:
> >     pyproj_datadir = os.environ['PROJ_DIR']
> > else:
> >     conda_file_dir = conda.CONDA_PACKAGE_ROOT
> >     conda_dir = conda_file_dir.split('lib')[0]
> >     pyproj_datadir = os.path.join(os.path.join(conda_dir, 'Library'),'share')
> > ```
> > 
> > 
> > The user just need to check the path to where the python system saved the epsg of pyproj/proj,
> > which is usually in Library/share or inside the the pyproj folder.
> > For the record, I'm using win10, anaconda 5.3.1, python 3.7.0 and I changed my first level channel to conda-forge.
> > regards
> 
> Seconding that after having wasted a couple hours on this, the only solution that worked for me also, was @rpedruzzi's. I'm running on Windows 10, Python 3.6.

Hi I was trying to follow the steps but i am getting ModuleNotFoundError: No module named 'conda'.

  File "C:/Users/Dipankar/Anaconda2/envs/t1/lib/site-packages/mpl_toolkits/basemap/__init__.py", line 2, in <module>
    import os,conda

ModuleNotFoundError: No module named 'conda'.

Can you please give me some suggestion on it

**Comment by Leasberthou** [comment]
You just have to add 'import conda' with the other imported packages in the top of the file.

**Comment by DipankarMedhi** [comment]
> You just have to add 'import conda' with the other imported packages in the top of the file.

I did that even after that I am getting the error. Do you want me to import conda in init file I have done so

**Comment by Leasberthou** [comment]
Weird, this worked for me. Do you have the conda package installed and have you checked if you can import conda in a normal python script?

**Comment by rpedruzzi** [comment]
Dear all,

I found out that the correct way to open Spyder in Win10 is opening the Anaconda prompt and open it using this command:
`spyder --hide-console`

the -- hide-console is to close the anaconda prompt after launch spyder.
When you do this, Spyder gets all environment variables, including the PROJ_LIB and all others.
This step are necessary because there some config scripts which is run when you open Anaconda prompt that creates all this variables, so Spyder can get it. Open it directly on the app icon will not assign these variables and paths causing errors.
I got this information in other Spyder issues comments by @ccordoba12 

Regards
